File: Competitive Programming 3/Section 1.3.3/11044 - Searching for Nessy/C++/main.cpp
#include <iostream>

using namespace std;

int extraDiv(int a, int b){
    return a%b?1:0;
}

int operation(int a){
    a-=2;
    return a/3 + extraDiv(a,3);
}

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    int t ; 
    cin >> t;
    while(t--){
        int n,m;
        cin >> n >> m ; 
        cout << operation(n) * operation(m) << "\n";
    }
    return 0;
}

File: Competitive Programming 3/Section 1.3.3/11044 - Searching for Nessy/Java/Main.java
import java.util.Scanner;

public class Main{
    static int extraDiv(int a,int b){
        return a%b>0? 1 : 0 ;
    }
    static int operation(int a){
        a-=2;
        //System.out.printf("a/3= %d\n",a/3);
        return a/3 + extraDiv(a,3);
    }
    public static void main(String[]args){
        Scanner s = new Scanner(System.in);
        for(int t = s.nextInt(); t > 0 ; t --){
            int n = s.nextInt(), m = s.nextInt();
            System.out.printf("%d\n",operation(n) * operation(m));
        }
    }
}